### PR TITLE
Add server and basic frontend

### PR DESCRIPTION
**Pull request by JakubC-projects**
Adds the server to the solution.
The server does 2 things:
1. Proxy requests at `/api` path to the core api
2. Serve frontend

Server uses auth0 to authenticate users, and maintains a cookie session with the browser

In this PR the server is fully done

### CODE REVIEW

File: server/api/api.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"golang.org/x/oauth2"
+)
+
+func Handler(remote string, tokens oauth2.TokenSource) http.Handler {
+	remoteUrl, err := url.Parse(remote)
+	if err != nil {
+		panic(err)
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		t, err := tokens.Token()
```
[the remark this answers is not in the document]
Comment (JakubC-projects) [reply server/api/api.go:19]: no, it's actually client credentials token of the BFF
tokens here is of type `oauth2.TokenSource`, reusing, and if necessary refetching the token

File: server/auth/middleware.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+		next.ServeHTTP(w, req)
+	})
+}
+
+func (a *Auth) ClaimCheck(verifyFunc func(s Session) error, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		session, err := a.getSession(req)
+		if err != nil {
+			w.WriteHeader(http.StatusUnauthorized)
```
[the remark this answers is not in the document]
Comment (JakubC-projects) [reply server/auth/middleware.go:24]: So, in the session cookie there is ExpiresAt field, so if the session is expired, then we do not allow an action to be performed

File: server/cmd/server/main.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		cfg.Coreapi.BasePath,
+		getCoreApiTokenSource(a, cfg.Coreapi.Audience)))
+	authenticatedMux.Handle("/", (frontend.Handler(cfg.FrontendLocation)))
+
+	http.ListenAndServe(fmt.Sprintf(":%d", cfg.Server.Port), mux)
+}
+
+func getCoreApiTokenSource(a *auth.Auth, audience string) oauth2.TokenSource {
+	clientCredentialsCfg := &clientcredentials.Config{
+		ClientID:     a.Config().ClientId,
```
[the remark this answers is not in the document]
Comment (JakubC-projects) [reply server/cmd/server/main.go:48]: For client credentials flow `scope` parameter is optional

[the remark this answers is not in the document]
Comment (rvanoord) [reply server/cmd/server/main.go:48]: We should always specify scope for our services. I know Auth0 just adds all of them, but I don't think adding all of them is part of the "protocol". Also, minimizing scopes is in line with the principal of least privilege. @JakubC-projects

**Pull request merged**